Source/NSPersonNameComponentsFormatter.m

```diff
@@ -0,0 +1,236 @@
+{
+  _nameOptions = opts;
+}
+
+// Designated init...
++ (NSString *) localizedStringFromPersonNameComponents: (NSPersonNameComponents *)components
+                                                 style: (NSPersonNameComponentsFormatterStyle)nameFormatStyle
+                                               options: (NSPersonNameComponentsFormatterOptions)nameOptions
+{
+  NSPersonNameComponentsFormatter *fmt = [[NSPersonNameComponentsFormatter alloc] init];
+  [fmt setStyle: nameFormatStyle];
+  [fmt _setNameOptions: nameOptions];
+  return [fmt stringForObjectValue: components] ;
+}
+
+// Setters
+- (NSPersonNameComponentsFormatterStyle) style
+{
+  return _style;
+}
+
+- (void) setStyle: (NSPersonNameComponentsFormatterStyle)style
+{
+  _style = style;
+}
+
+- (BOOL) isPhonetic
+{
+  return _phonetic;
+}
+
+- (void) setPhonetic: (BOOL)flag
+{
+  _phonetic = flag;
+}
+
+// Convenience methods...
+- (NSString *) stringFromPersonNameComponents: (NSPersonNameComponents *)components
+{
+  NSString *result = @"";
+  
+  switch (_style)
+    {
+    case NSPersonNameComponentsFormatterStyleDefault:
+    case NSPersonNameComponentsFormatterStyleMedium:
+      result = [result stringByAppendingString: [components givenName]];
+      result = [result stringByAppendingString: @" "];
+      result = [result stringByAppendingString: [components familyName]];
+      break;
+    case NSPersonNameComponentsFormatterStyleShort:
+      result = [result stringByAppendingString: [components givenName]];
+      break;
+    case NSPersonNameComponentsFormatterStyleLong:      
+      result = [result stringByAppendingString: [components namePrefix]];
+      result = [result stringByAppendingString: @" "];
+      result = [result stringByAppendingString: [components givenName]];
+      result = [result stringByAppendingString: @" "];
+      result = [result stringByAppendingString: [components familyName]];
+      result = [result stringByAppendingString: @" "];
+      result = [result stringByAppendingString: [components nameSuffix]];
+      break;
+    case NSPersonNameComponentsFormatterStyleAbbreviated:
+      result = [result stringByAppendingString: [[components givenName] substringToIndex: 1]];
+      result = [result stringByAppendingString: [[components familyName] substringToIndex: 1]];
+      break;
+    }
+  
+  return result;
+}
+
+- (NSAttributedString *) annotatedStringFromPersonNameComponents: (NSPersonNameComponents *)components
+{
+  NSAttributedString *result =  AUTORELEASE([[NSAttributedString alloc] initWithString:
+                                                  [self stringFromPersonNameComponents: components]]);
+  return result;
+}
+
+- (NSPersonNameComponents *) personNameComponentsFromString: (NSString *)string
+{
+  NSPersonNameComponents *pnc = AUTORELEASE([[NSPersonNameComponents alloc] init]);
+  NSArray *nameArray = [string componentsSeparatedByString: @" "];
+  NSUInteger count = [nameArray count];
+
+  switch(count)
```

Comment:
Shouldn't this method look at the _style variable?
At least it should set the given name in the first case to match the formatting above.

Reply:
I had assumed that it doesn't only because the documentation makes no mention of it.  The logic of the method is fairly simple without it.   Additionally I think style should effect output and not parsing.